libs/vm_modules/tests/unit/fixed_point.cpp

```diff
@@ -28,21 +30,173 @@ class FixedPointTest : public ::testing::Test
   VmTestToolkit     toolkit{&stdout};
 };
 
-TEST_F(FixedPointTest, create_fixed_point)
+bool RunTest(VmTestToolkit &toolkit, std::stringstream &stdout, char const *TEXT, double gt)
 {
   auto m = toolkit.module();
+  EXPECT_TRUE(toolkit.Compile(TEXT));
+  EXPECT_TRUE(toolkit.Run());
+  EXPECT_EQ(std::stod(stdout.str()), gt);
+  return true;
+}
 
-  static char const *TEXT = R"(
+TEST_F(FixedPointTest, create_fixed_point)
+{
+  char const *TEXT = R"(
     function main()
       print(1.0fp32);
     endfunction
   )";
+  double      gt   = static_cast<double>(fetch::fixed_point::fp32_t(1));
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+}
 
-  ASSERT_TRUE(toolkit.Compile(TEXT));
-  ASSERT_TRUE(toolkit.Run());
+TEST_F(FixedPointTest, addition_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var a = 2.0fp32;
+      var b = 3.0fp32;
+      a += b;
+      print(a);
+    endfunction
+  )";
+  double      gt   = static_cast<double>(fetch::fixed_point::fp32_t(5));
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+}
 
-  double gt = static_cast<double>(fetch::fixed_point::fp32_t(1));
-  EXPECT_EQ(std::stod(stdout.str()), gt);
+TEST_F(FixedPointTest, subtraction_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var a = 3.0fp32;
+      var b = 2.0fp32;
+      a -= b;
+      print(a);
+    endfunction
+  )";
+  double      gt   = static_cast<double>(fetch::fixed_point::fp32_t(1));
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+}
+
+TEST_F(FixedPointTest, multiplication_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var a = 3.0fp32;
+      var b = 2.0fp32;
+      a *= b;
+      print(a);
+    endfunction
+  )";
+  double      gt   = static_cast<double>(fetch::fixed_point::fp32_t(6));
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+}
+
+TEST_F(FixedPointTest, divide_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var a = 3.0fp32;
+      var b = 2.0fp32;
+      a /= b;
+      print(a);
+    endfunction
+  )";
+  double      gt   = static_cast<double>(fetch::fixed_point::fp32_t(1.5));
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
 }
 
+TEST_F(FixedPointTest, array_32_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var myArray = Array<Fixed32>(5);
+
+      for (i in 0:4)
+        myArray[i] = toFixed32(i);
+      endfor
+      print(myArray[3]);
+    endfunction
+  )";
+  double      gt   = static_cast<double>(fetch::fixed_point::fp32_t(3));
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+}
+
+TEST_F(FixedPointTest, array_64_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var myArray = Array<Fixed64>(5);
+
+      for (i in 0:4)
+        myArray[i] = toFixed64(i);
+      endfor
+      print(myArray[3]);
+    endfunction
+  )";
+  double      gt   = static_cast<double>(fetch::fixed_point::fp64_t(3));
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+}
+//
+// TEST_F(FixedPointTest, map_32_fixed_point)
```

Comment:
This is not good practice - commented out tests are easy to forget about. This test looks reasonable. If the underlying functionality is not implemented yet, uncomment the test and call it `DISABLED_map_32_fixed_point`, then gtest will ignore it and report that there's a disabled test.